packages/explat-client/src/internal/timing.ts

```diff
@@ -0,0 +1,57 @@
+export const MILLISECONDS_PER_SECOND = 1000;
+
+let lastNow = Date.now();
+/**
+ * Returns the time in miliseconds.
+ * A strictly increasing version of Date.now()
+ */
+export function monotonicNow(): number {
+	const maybeNow = Date.now();
+
+	lastNow = lastNow < maybeNow ? maybeNow : lastNow + 1;
+
+	return lastNow;
+}
+
+export function createUnresolvingPromise< T >(): Promise< T > {
+	return new Promise< T >( () => {
+		return;
+	} );
+}
+
+/**
+ * Timeouts a promise. Returns timeoutValue in event of timeout.
+ *
+ * @param promise The promise to timeout
+ * @param ms The timeout time in milliseconds
+ */
+export function timeoutPromise< T >( promise: Promise< T >, ms: number ): Promise< T | null > {
```

Comment:
This sounds like it time outs the promise, but it actually creates a promise that would time out after `ms`. Maybe call it `addPromiseTimeout` or `setPromiseTimeout` and update the documentation accordingly?

Also, `timeoutMilliseconds` or `timeoutMs` would be a bit more informative than `ms`.

Reply:
> This sounds like it time outs the promise

Isn't this timing out a promise? I'm curious what that means to you?

> Also, timeoutMilliseconds or timeoutMs would be a bit more informative than ms.

👍